plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftUnloadSplitSource.java

```diff
@@ -0,0 +1,196 @@
+    }
+
+    @Override
+    public CompletableFuture<ConnectorSplitBatch> getNextBatch(int maxSize)
+    {
+        return resultSetFuture
+                .thenApply(_ -> {
+                    ConnectorSplitBatch connectorSplitBatch = new ConnectorSplitBatch(readUnloadedFilePaths().stream()
+                            .map(fileInfo -> (ConnectorSplit) new RedshiftUnloadSplit(fileInfo.path, fileInfo.size))
+                            .collect(toImmutableList()), true);
+                    finished = true;
+                    return connectorSplitBatch;
+                });
+    }
+
+    @Override
+    public void close()
+    {
+        resultSetFuture.cancel(true);
+    }
+
+    @Override
+    public boolean isFinished()
+    {
+        return finished;
+    }
+
+    private String buildRedshiftSelectSql(ConnectorSession session, Connection connection, JdbcTableHandle table, List<JdbcColumnHandle> columns)
+            throws SQLException
+    {
+        PreparedQuery preparedQuery = jdbcClient.prepareQuery(session, table, Optional.empty(), columns, ImmutableMap.of());
+        String selectQuerySql;
+        try (PreparedStatement openTelemetryPreparedStatement = queryBuilder.prepareStatement(jdbcClient, session, connection, preparedQuery, Optional.of(columns.size()))) {
+            RedshiftPreparedStatement redshiftPreparedStatement = openTelemetryPreparedStatement.unwrap(RedshiftPreparedStatement.class);
+            selectQuerySql = redshiftPreparedStatement.toString();
+        }
+        return queryModifier.apply(session, selectQuerySql);
+    }
+
+    private PreparedStatement buildRedshiftUnloadSql(ConnectorSession session, Connection connection, List<JdbcColumnHandle> columns, String redshiftSelectSql, String unloadOutputPath)
+            throws SQLException
+    {
+        String unloadSql = "UNLOAD ('%s') TO '%s' IAM_ROLE %s FORMAT PARQUET MAXFILESIZE 64MB MANIFEST VERBOSE".formatted(
+                escapeUnloadIllegalCharacters(redshiftSelectSql),
+                unloadOutputPath,
+                unloadAuthorization.map("'%s'"::formatted).orElse("DEFAULT"));
+        return queryBuilder.prepareStatement(jdbcClient, session, connection, new PreparedQuery(unloadSql, List.of()), Optional.of(columns.size()));
+    }
+
+    private List<FileInfo> readUnloadedFilePaths()
+    {
+        Location manifestLocation = Location.of(unloadOutputPath + "manifest");
+        TrinoInputFile inputFile = fileSystem.newInputFile(manifestLocation);
+        JsonNode outputFileEntries;
+        try (TrinoInputStream inputStream = inputFile.newStream()) {
+            byte[] manifestContent = inputStream.readAllBytes();
+            outputFileEntries = OBJECT_MAPPER.readTree(manifestContent).path("entries");
+        }
+        // manifest is not generated if unload query doesn't produce any results.
+        // Rely on the catching `FileNotFoundException` for the absence of manifest file as `TrinoInputFile#exists` adds additional call to S3.
```

Comment:
Check the existence of the file with `inputFile.exists()` instead of relying on `FileNotFoundException`.

Add a test on `TestRedshiftUnload` for dealing with an empty resultset.

Reply:
>Check the existence of the file with inputFile.exists() instead of relying on FileNotFoundException.

Copying the code comment:
>        // Rely on the catching `FileNotFoundException` for the absence of manifest file as `TrinoInputFile#exists` adds additional call to S3.

It would be better to provide more clarification on why inputFile.exists() enhances performance compared to checking exception.

I've added a test that verifies that this logic works as expected.

Reply:
I'm not strong on changing the code. It was a suggestion.